scala/scalafmt/BUILD

```diff
@@ -0,0 +1,37 @@
+load("//scala:scala.bzl", "scala_binary")
+
+filegroup(
+    name = "runner",
+    srcs = ["private/format.template.sh"],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "testrunner",
+    srcs = ["private/format-test.template.sh"],
+    visibility = ["//visibility:public"],
+)
+
+scala_binary(
+    name = "scalafmt",
+    srcs = glob(["scalafmt/ScalafmtRunner.scala"]),
+    main_class = "io.bazel.rules_scala.scalafmt.ScalafmtRunner",
+    visibility = ["//visibility:public"],
```

Comment:
do you want users to be able to use this target?

Reply:
They can if they want. I can't think of a strong reason to forbid it. What do you think?

Reply:
I usually try to be explicit about what I expose as a public API. Mainly since if we change this and people depend on it then they'd be surprised the contract changed. 
If we have a good reason then fine, otherwise maybe change to `//...` equivalent (which means only this repo can use this target). If there is a good reason then we should discuss.

WDYT?

Reply:
Sounds reasonable, but there is a challenge. Every target that uses format extension will need access to this binary, it could also lives in users' private workspace. Could it be not public?

Reply:
I'm pretty sure this isn't a problem. Visibility doesn't effect transitive dependencies but only who can directly depend on it.

Reply:
Sorry, it might be obvious to you but I can't get it to work if not public.

I got the message
```
ERROR: in scalafmt_scala_binary rule //test/scalafmt:formatted-binary: target '//scala/scalafmt:scalafmt' is not visible from target '//test/scalafmt:formatted-binary'. Check the visibility declaration of the former target if you think the dependency is legitimate
```
I appreciate if you can point the problem out.

Reply:
It seems I was wrong. I'm not really sure why but let's ignore for now so we can get this in